hal/src/sercom/v2/uart/reg.rs

```diff
@@ -0,0 +1,400 @@
+    #[inline]
+    fn usart(&self) -> &pac::sercom0::USART {
+        self.sercom.usart()
+    }
+
+    /// Helper function to access the underlying `USART_INT` from the given
+    /// `SERCOM`
+    #[cfg(feature = "min-samd51g")]
+    #[inline]
+    fn usart(&self) -> &pac::sercom0::USART_INT {
+        self.sercom.usart_int()
+    }
+
+    #[cfg(feature = "dma")]
+    /// Get a pointer to the `DATA` register
+    pub(super) fn data_ptr<T>(&self) -> *mut T {
+        self.usart().data.as_ptr() as *mut _
+    }
+
+    /// Free the `Registers` struct and return the underlying `Sercom` instance
+    #[inline]
+    pub(super) fn free(self) -> S {
+        self.sercom
+    }
+
+    /// Reset the SERCOM peripheral
+    #[inline]
+    pub(super) fn swrst(&mut self) {
+        self.usart().ctrla.write(|w| w.swrst().set_bit());
+        while self.usart().syncbusy.read().swrst().bit_is_set() {}
+    }
+
+    /// Configure the SERCOM to use internal clock mode
+    #[inline]
+    pub(super) fn configure_mode(&mut self) {
+        self.usart()
+            .ctrla
+            .modify(|_, w| w.mode().variant(MODE_A::USART_INT_CLK));
+    }
+
+    /// Configure the `SERCOM`'s Pads according to RXPO and TXPO
+    #[inline]
+    pub(super) fn configure_pads(&mut self, rxpo: u8, txpo: u8) {
+        self.usart().ctrla.modify(|_, w| unsafe {
+            w.rxpo().bits(rxpo);
+            w.txpo().bits(txpo)
+        });
+    }
+
+    /// Configure the character size
+    #[inline]
+    pub(super) fn configure_charsize(&mut self, bits: u8) {
+        self.usart()
+            .ctrlb
+            .modify(|_, w| unsafe { w.chsize().bits(bits) });
+    }
+
+    /// Change the bit order of transmission (MSB/LSB first)
+    #[inline]
+    pub(super) fn set_bit_order(&mut self, bit_order: BitOrder) {
+        self.usart()
+            .ctrla
+            .modify(|_, w| w.dord().bit(bit_order.into()));
+    }
+
+    /// Get the current bit order
+    #[inline]
+    pub(super) fn get_bit_order(&self) -> BitOrder {
+        self.usart().ctrla.read().dord().bit().into()
+    }
+
+    /// Change the parity setting
+    #[inline]
+    pub(super) fn set_parity(&mut self, parity: Parity) {
+        // Use only the first two available settings in the FORM field.
+        // Ignore auto-baud options.
+        let enabled = match parity {
+            Parity::None => false,
+            Parity::Odd => {
+                self.usart().ctrlb.modify(|_, w| w.pmode().bit(true));
+                true
+            }
+            Parity::Even => {
+                self.usart().ctrlb.modify(|_, w| w.pmode().bit(false));
+                true
+            }
+        };
+
+        self.usart()
+            .ctrla
+            .modify(|_, w| unsafe { w.form().bits(!enabled as u8) });
+    }
+
+    /// Get the current parity setting
+    #[inline]
+    pub(super) fn get_parity(&self) -> Parity {
+        let enabled = self.usart().ctrla.read().form().bits() != 0;
```

Comment:
Just thinking about our future selves, it might be better if this checked for the format being either 0x1 (USART w/ parity) or 0x5 (auto baud w/ parity).  If auto-baud w/o parity were enabled (form 0x4), this would return the wrong result.